pkg/model/firewall.go

```diff
@@ -226,56 +227,134 @@ func (b *FirewallModelBuilder) applyNodeToMasterBlockSpecificPorts(c *fi.ModelBu
 		protocols = append(protocols, ProtocolIPIP)
 	}
 
-	for _, r := range udpRanges {
-		c.AddTask(&awstasks.SecurityGroupRule{
-			Name:          s(fmt.Sprintf("node-to-master-udp-%d-%d", r.From, r.To)),
-			Lifecycle:     b.Lifecycle,
-			SecurityGroup: b.LinkToSecurityGroup(kops.InstanceGroupRoleMaster),
-			SourceGroup:   b.LinkToSecurityGroup(kops.InstanceGroupRoleNode),
-			FromPort:      i64(int64(r.From)),
-			ToPort:        i64(int64(r.To)),
-			Protocol:      s("udp"),
-		})
+	for masterSecGroupName, masterGroup := range masterGroups {
+
+		if len(masterGroups) != 1 {
+			masterSecGroupName = "-" + masterSecGroupName
+		} else {
+			masterSecGroupName = ""
+		}
+
+		for nodeSecGroupName, nodeGroup := range nodeGroups {
+
+			if len(masterGroups) == 1 && len(nodeGroups) == 1 {
+				nodeSecGroupName = ""
+			} else {
+				nodeSecGroupName = fmt.Sprintf("-%s-%s", masterSecGroupName, nodeSecGroupName)
```

Comment:
This looks like we'll have a `--` in there

Reply:
I will test - but the test was suppose to pick up wen we need to suffix .... the logic is not clean